client/List.jsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class Item extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {isCompleted: false};
+        this.handleClick = this.handleClick.bind(this);
+    }
+    handleClick(){
```

Comment:
`handleClick` can be passed in from the parent as a prop that updates the parent state.  This way `Item` can be a stateless functional component.

Reply:
I get the idea of stateless components but for this case wouldn't we want to have the event handler in the component itself. I thought in React we should try to avoid manipulating the parent's state?

Reply:
Manipulating the parent's state is totally fine and especially when writing a small application like this where we're only drilling props down one layer.  It also prevents us from being bogged down by instances of `Component`, and keeping state in multiple places in our app.

General you want to use individual component state for component-specific state handling.  Like if it's a drawer component it could be whether or not a drawer is open.  Maybe a dropdown saying whether or not the dropdown is open.  These are perfectly reasonable places to use component state.

Application state on the other hand may be what value is selected in the dropdown, or what is being updated in the drawer.  Essentially the data elements.  

Passing down a stateful function into a child component will also make this a bit more extensible, because it is asserting a convention that is easy to follow.  No one will look at that code and be confused, and it'll be easy to make the state do something a little more concrete like fire a REST request that actually calls a DELETE method or something.